### DISCUSSION TITLE
Lua Autosplitter Context

### DISCUSSION THREAD

**Discussion by delcake** [wins1ey/LibreSplit#31] [Q&A] [answered]
After quite a lot of trial and error, I've started to have some success with an autosplitter script I'm working on thanks to the newly available documentation, so thanks for making that available! I'm curious whether the autosplitter Lua context has access to information about the current split, specifically the current split's title. If so, how would I be able to reference the current split's title in my script?

For context of why I'm asking, I activate a split based on when the player's chat log reports a quest as having been completed. While I could obtain the name of the quest to check against from the game's memory, the quest names that I want to be able to trigger a split are already included as split names. So it seems like a good way to simplify my script if this information is already obtainable directly from LAST.

**Comment by Loomeh** [comment, marked as answer]
Hi,

LAST's Lua context by itself does not have access to any info about splits. All of LAST's current Lua functions are listed in the documentation.

Even if it was able to read the split title, it probably wouldn't be a good idea to use it any way as it forces runners to have exact split names. It would probably be better to search for a quest index or the current quest's name in memory.

**Reply by delcake** [reply]
Thanks for confirming! Yeah, I agree that in a situation where the autosplitter would have more users than just myself a more robust design that doesn't require exact split names would definitely be preferable.
